internal/restorer/filerestorer.go

```diff
@@ -268,7 +278,15 @@ func (r *fileRestorer) downloadPack(ctx context.Context, pack *packInfo) error {
 						file.inProgress = true
 						createSize = file.size
 					}
-					return r.filesWriter.writeToFile(r.targetPath(file.location), blobData, offset, createSize, file.sparse)
+					writeErr := r.filesWriter.writeToFile(r.targetPath(file.location), blobData, offset, createSize, file.sparse)
+
+					if r.printProgress {
+						bytesWrittenPortion := int64(len(blobData))
+						progress := r.progressFormatter.FormatProgress(file.location, bytesWrittenPortion, file.size)
+						r.terminal.SetStatus([]string{progress})
```

Comment:
setting the status belongs into the progressFormatter. The fileRestorer should only report what has changed. But the task of transforming that into a status output belongs into the progressFormatter. Thus, please change that back to your initial implementation.